firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/Containers/EnumMap.txt

```diff
@@ -0,0 +1,357 @@
+SCENARIO('The EnumMap is constructed from initializer list works corretly') 
+  GIVEN('An EnumMap constructed with capacity 11, has {1, 1}, {2,2}, {3,3}, {4,4} items') 
+    WHEN('The size method is called')
+      THEN('The size method returns size as 4')
+
+    WHEN('The avaliable method is called ')
+      THEN('The avaliable method reports that 7 items are avlaible ')
+
+    WHEN('The full method is called')
+      THEN('The full method reports that the EnumMap is not full')
+
+    WHEN('The empty method is called')
+      THEN('The empty method reports that the EnumMap is not empty')
+
+    WHEN('The has method is called')
+      THEN('The has method returns true for 1-4 keys')
+      THEN('The has method returns false for [-1000]to[1000] keys')
+
+    WHEN('Indexing thorugh EnumMap')
+      THEN('Indexing into the containers for keys 1, 2, 3, 4 returns the values 1, 2, 3, 4 respectively')
+
+    WHEN('The input method is called on 5th item with value as 5') 
```

Comment:
The EnumMap doesn't really have the concept of an "nth item". It would be clearer and more consistent to say that the input method is called with key 5 and value 5 (assuming that's what you meant).